splunk/tests/splunk_test_prep.go

```diff
@@ -0,0 +1,73 @@
+package main
+
+import (
+	"io/ioutil"
+	"log"
+	"os"
+	"os/exec"
+	"strings"
+	"time"
+)
+
+// BuildSplunk builds the Splunk image used for testing Splunk queries.
+func BuildSplunk() {
+	buildCmd := exec.Command("podman", "build", "-t", "test_splunk", "./splunk")
+
+	buildCmd.Stdout = os.Stdout
+	buildCmd.Stderr = os.Stderr
+	err := buildCmd.Run()
+	if err != nil {
+		log.Fatalf("Failed to build image: %v", err)
+	}
+	log.Println("Image build done successfully!")
+}
+
+// RunSplunk deploy the Splunk container used for testing Splunk queries.
+func RunSplunk() {
```

Comment:
We probably need a way to stop it too... I wonder if `go test` has a way to have session-wide fixtures

Reply:
If we have the container ID, maybe we can stop it during teardown, using TestMain https://medium.com/goingogo/why-use-testmain-for-testing-in-go-dafb52b406bc ?

Reply:
the problem with having a fixed container ID si that the user may already have something there. We need to stay out of the user's way as much as possible.

Reply:
Not sure I get it. how's the container ID fixed if we can get it when first creating it?

Reply:
I was thinking you mean a container name. In any case, for any resource that is in a limited, shared namespace such as port numbers, image names, container names, etc. We must make sure we never collide with something the user may already have there, including another parallel run of our own code.

As far as cleanup goes - I thought about this and realize we' can probably have a single test with multiple subtests, so something along the lines of what I've made for Segment where we have a function that defers the cleanup and calls the test code in a function passed to it may work just as well for this case.

Reply:
I'll try implementing it, 
for now we'll have to verify the port number at the very least. We are currently getting the Container ID when running the container, so at least that's dealt with

Reply:
Instead of setting image or container names, I'm having podman setting them up randomly so we won't have any issues with those shared namespaces. 
But checking the port number introduces a race condition between the check and running the container...